libs/atqr/domain/src/lib/notifications/notification.valueObject.ts

```diff
@@ -0,0 +1,109 @@
+import { ValueObject } from '../../utils/valueObject';
+import { Guid } from '@tokilabs/lang';
+import { NotificationLogEntry } from './notificationLogEntry.valueObject';
+import { isBeforeDay } from '../../utils/isBeforeDay';
+
+enum NotificationCategory {
+  EuDuvidoProductUpdates = 'EuDuvidoProductUpdates',
+  EuDuvidoPromotions = 'EuDuvidoPromotions',
+  NewInvitations = 'NewInvitations',
+  ChallengeUpdates = 'ChallengeUpdates',
+}
+
+export enum NotificationChannel {
+  Email = 'Email',
+  SMS = 'SMS',
+  Push = 'Push',
+}
+
+export class Notification extends ValueObject<Notification> {
+  constructor(
+    public readonly category: NotificationCategory,
+    public readonly templateId: MessageTemplateId,
+    public readonly allowedChannels: NotificationChannel[],
+    public readonly createdAt: Date,
+    public readonly requiresUserAction: boolean,
+    public readonly actionTaken: boolean,
+    public readonly contactLog: NotificationLogEntry[],
+    public readonly nextContactDate: Date,
+    public readonly id = new Guid()
```

Comment:
the "id" property should actually be the first one in the constructor, if you follow the pattern used until here.